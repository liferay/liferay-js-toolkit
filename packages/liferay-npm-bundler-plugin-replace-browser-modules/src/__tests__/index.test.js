import cpFile from 'cp-file';
import * as fs from 'fs';
import plugin from '../index';

// Package descriptor used in tests
const pkg = {
	id: 'package@1.0.0',
	name: 'package',
	version: '1.0.0',
	dir:
		`${process.cwd()}/packages/` +
		`liferay-npm-bundler-plugin-replace-browser-modules/src/__tests__`,
};

// Package files used in tests
const pkgFiles = [
	'test-2.js.file',
	'test-browser-2.js.file',
	'test-browser.js.file',
	'test-main.js.file',
	'test.js.file',
];

// Restore package status prior to running each test
beforeEach(() => {
	pkgFiles.forEach(file => {
		cpFile.sync(`${pkg.dir}/sources/${file}`, `${pkg.dir}/${file}`);
	});
});

// Delete result files after running each test
afterEach(() => {
	pkgFiles.forEach(file => {
		fs.unlink(`${pkg.dir}/${file}`);
	});
});

it('replaces main file with browser file', () => {
	const pkgJson = {
		name: pkg.name,
		version: pkg.version,
		main: 'test-main.js.file',
		browser: 'test-browser.js.file',
	};

	plugin({ pkg }, { pkgJson });

	expect(
		fs.readFileSync(`${pkg.dir}/test-main.js.file`).toString(),
	).toMatchSnapshot();
});

it('replaces server files with browser files', () => {
	const pkgJson = {
		name: pkg.name,
		version: pkg.version,
		browser: {
			'test-browser.js.file': 'test.js.file',
			'test-browser-2.js.file': 'test-2.js.file',
		},
	};

	plugin({ pkg }, { pkgJson });

	expect(
		fs.readFileSync(`${pkg.dir}/test-browser.js.file`).toString(),
	).toMatchSnapshot();
	expect(
		fs.readFileSync(`${pkg.dir}/test-browser-2.js.file`).toString(),
	).toMatchSnapshot();
});

it('does replace ignored modules with empty objects', () => {
	const pkgJson = {
		name: pkg.name,
		version: pkg.version,
		main: 'main.js',
		browser: {
			'test-browser.js.file': false,
		},
	};

	plugin({ pkg }, { pkgJson });

	expect(
		fs.readFileSync(`${pkg.dir}/test-browser.js.file`).toString(),
	).toMatchSnapshot();
});

it('does not throw on inexistent files', () => {
	const pkgJson = {
		name: pkg.name,
		version: pkg.version,
		main: 'main.js',
		browser: 'non-existent-file.js',
	};

	plugin({ pkg }, { pkgJson });
});
