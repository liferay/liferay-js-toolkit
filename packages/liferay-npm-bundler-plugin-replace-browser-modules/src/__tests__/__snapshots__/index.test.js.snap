// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does replace ignored modules with empty objects 1`] = `
"/* Module ignored by liferay-npm-bundler-plugin-replace-browser-modules */
"
`;

exports[`replaces main file with browser file 1`] = `
"/* Module replaced with test-browser.js.file by liferay-npm-bundler-plugin-replace-browser-modules */
/**
 * testBrowser function
 * @return {void}
 */
export default function testBrowser() {
	console.log('Hello from testBrowser');
}
"
`;

exports[`replaces server files with browser files 1`] = `
"/* Module replaced with test.js.file by liferay-npm-bundler-plugin-replace-browser-modules */
/**
 * test function
 * @return {void}
 */
export default function test() {
	console.log('Hello from test');
}
"
`;

exports[`replaces server files with browser files 2`] = `
"/* Module replaced with test-2.js.file by liferay-npm-bundler-plugin-replace-browser-modules */
/**
 * test2 function
 * @return {void}
 */
export default function test2() {
	console.log('Hello from test2');
}
"
`;
